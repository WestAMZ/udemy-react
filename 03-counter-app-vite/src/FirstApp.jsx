
import PropTypes from 'prop-types'

export const FirstApp = ( { title, subTitle, name }) => {

  return (
    <>
        <h1 data-testid="test-title">{title}</h1>
        <p>{subTitle}</p>
        <p>{subTitle}</p>
        <p>{name}</p>
    </>
    
  )
}

FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subTitle: PropTypes.string
}

// Default props entran antes que los propTypes es decir
// si no se pasó un valor pero se definió en defaultProps
// no tendremos error de propTypes
FirstApp.defaultProps = {
    //title: 'No hay titulo',
    name: 'Fernando Herrera',
    subTitle: 'No hay subtitulo'
}