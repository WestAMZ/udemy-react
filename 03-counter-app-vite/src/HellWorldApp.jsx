export const HellWorldApp = () => {
  return (
    <h1>HelloWorld</h1>
  )
}
// shorcut para crear componente funcional
// rafc