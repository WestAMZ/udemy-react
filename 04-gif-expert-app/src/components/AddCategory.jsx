import { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({onNewCategory}) => {

    const [inputValue, setInputValue] = useState('')

    const onInputChange = ({target}) => {
        setInputValue(target.value);
    }

    const onSubmit = (event) => {
        event.preventDefault();

        if(inputValue.trim().length <= 1) return;
        
        onNewCategory( inputValue.trim() );
        setInputValue('');
    }

    return (
        // Atributo aria-label="form", agregadp àra ser reconocido por react-testing library
        <form onSubmit={ onSubmit } aria-label="form">
            <input
                type= "text"
                placeholder="Buscar Gif"
                value = {inputValue}
                //onChange={ (event) => onInputChange(event)
                // parametro event es pasado a la función en automático
                onChange={ onInputChange }
            />
        </form>
    )
}

AddCategory.propTypes = {
    onNewCategory : PropTypes.func.isRequired
}