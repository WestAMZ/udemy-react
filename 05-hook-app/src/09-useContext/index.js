
export * from './AboutPage.jsx';
export * from './HomePage.jsx';
export * from './LoginPage.jsx';