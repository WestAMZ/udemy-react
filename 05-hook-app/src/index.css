body {
    padding: 20px;
}

button{
    margin-right: 10px;
}

.bg-normal{
    background-color: rgb(160, 160, 160);
}

.bg-fire{
    background-color: rgb(226, 26, 0);
}

.bg-water{
    background-color: rgb(29, 107, 252);
}

.bg-grass{
    background-color: rgb(0, 216, 47);
}

.bg-poison{
    background-color: rgb(109, 0, 136);
}

.bg-bug{
    background-color: rgb(105, 136, 64);
}

.bg-flying{
    background-color: rgb(185, 124, 255);
}

.bg-ground{
    background-color: rgb(148, 150, 64);
}

.bg-fairy{
    background-color: rgb(204, 135, 198);
}

.bg-fighting{
    background-color: rgb(238, 88, 2);
}

.bg-psychic{
    background-color: rgb(255, 68, 239);
}