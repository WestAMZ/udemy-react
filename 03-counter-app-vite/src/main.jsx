import React from 'react';
import ReactDOM from 'react-dom/client'
// import { HellWorldApp } from './HellWorldApp';
import { FirstApp } from './FirstApp';

// Css styles import
import './styles.css'

ReactDOM.createRoot(document.getElementById('root'))
.render(
    <React.StrictMode>
        <FirstApp />
    </React.StrictMode>
);