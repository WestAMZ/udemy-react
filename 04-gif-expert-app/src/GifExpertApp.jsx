import { useState } from "react";
import { AddCategory } from "./components/AddCategory";
import { GifGrid } from "./components/GifGrid";

// luego de versión 17 de react no es necesario hacer la importación de Reac from ...
export const GifExpertApp = () =>{

    const [categories, setcategories] = useState(['One Punch'])

    const onAddCategory = (newCategory) => {
        if(categories.includes(newCategory)) return;

        setcategories( prevCategories =>[ newCategory, ...prevCategories ] );
    }

    return (
        <>
            <h1>GIfExpertApp</h1>
            <AddCategory 
                onNewCategory={ (value) => onAddCategory(value) }
            />

            { 
                categories.map(category => (
                    <GifGrid key={category} category={category} />
                ))
            }
        </>
    )
}