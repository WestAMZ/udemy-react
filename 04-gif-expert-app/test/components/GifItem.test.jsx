import { render, screen } from '@testing-library/react'
import { GifItem } from '../../src/components'

describe('Pruebas en <GifItem />', () =>{
    const title = 'Pruebas en <GifItem />'
    const url = 'https://media4.giphy.com/media/VXJWhaO7afRe/giphy.gif?cid=7f1ac2f6oze7n7tem6i9bmu0e0m0skhb896wn9mhtioo9mru&ep=v1_gifs_search&rid=giphy.gif&ct=g'

    test('debe de hacer match con snapshot', ()=>{

        const { container } = render(<GifItem title={title} url={url} />);
        expect(container).toMatchSnapshot();

    })

    test('debe mostrar la imagen con el URL y el ALT indicado', ()=>{

        render(<GifItem title={title} url={url} />);
        
        // expect( screen.getByRole('img').src).toBe(url);
        // expect( screen.getByRole('alt').alt).toBe(title);

        const { src, alt } = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);
        
    })

    test('debe mostrar el titulo en el componente', ()=>{

        render(<GifItem title={title} url={url} />);

        expect(screen.getByText( title )).toBeTruthy();
    })
})