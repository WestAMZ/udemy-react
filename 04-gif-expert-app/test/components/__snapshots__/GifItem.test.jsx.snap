// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem /> debe de hacer match con snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Pruebas en <GifItem />"
      src="https://media4.giphy.com/media/VXJWhaO7afRe/giphy.gif?cid=7f1ac2f6oze7n7tem6i9bmu0e0m0skhb896wn9mhtioo9mru&ep=v1_gifs_search&rid=giphy.gif&ct=g"
    />
    <p>
      Pruebas en &lt;GifItem /&gt;
    </p>
  </div>
</div>
`;
